pkg/result/result.go

```diff
@@ -20,13 +20,17 @@ type Result interface {
 	StringWithIndent(indent string) string
 	GetReason() Reason
 	ToError() error
+	Name() string
+	GetResult() map[string]interface{}
```

Comment:
I recommend to create an interface for the value of map and return a map of strongly defined interfaces. Also, I don't see where it is used.